packages/x-components/src/plugins/__tests__/x-plugin-mixin.spec.ts

```diff
@@ -24,6 +25,66 @@ describe('testing $x component API global mixin', () => {
     jest.clearAllMocks();
   });
 
+  describe('testing origin', () => {
+    it('includes "default" origin in the $x.emit metadata', () => {
+      const listener = jest.fn();
+
+      componentInstance.vm.$x.on('UserIsTypingAQuery', true).subscribe(listener);
+      componentInstance.vm.$x.emit('UserIsTypingAQuery', 'So awesome, much quality, such skill');
+
+      expect(listener).toHaveBeenCalledTimes(1);
+      expect(listener).toHaveBeenCalledWith({
+        eventPayload: 'So awesome, much quality, such skill',
+        metadata: { moduleName: null }
+      });
+    });
+
+    it('overrides the origin in the $x.emit metadata', () => {
+      const listener = jest.fn();
+
+      @Component({
+        template: `<div><slot/></div>`,
+        provide: {
+          origin: 'origin-test'
+        }
+      })
+      class Provider extends Vue {}
+
+      @Component({
+        mixins: [xComponentMixin(searchBoxXModule)]
+      })
+      class Injecter extends Vue {
+        render(createElement: CreateElement): VNode {
+          return createElement();
+        }
+      }
+
+      const wrapper = mount(
+        {
+          template: `<Provider><Injecter /></Provider>`,
+          components: {
+            Provider,
+            Injecter
+          }
+        } as ComponentOptions<any> & ThisType<Vue>,
+        {
+          localVue
+        }
+      );
+
+      wrapper
+        .findComponent(Injecter)
+        .vm.$x.emit('UserIsTypingAQuery', 'So awesome, much quality, such skill');
+      componentInstance.vm.$x.on('UserIsTypingAQuery', true).subscribe(listener);
+
+      expect(listener).toHaveBeenCalledTimes(1);
+      expect(listener).toHaveBeenCalledWith({
```

Comment:
instead of checking the whole event param maybe just check the origin:

```
      expect(listener).toHaveBeenCalledTimes(1);
      const { metadata } = listener.mock.calls[0][0];
      expect(metadata.origin).toBe('origin-test');
```

This way if metadata changes (that will always change), this test doesn't break